File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.java
class CustomComparator implements Comparator<Pair<Integer, Integer>> {
    public int compare(Pair<Integer, Integer> a, Pair<Integer, Integer> b) {
        if (!a.getKey().equals(b.getKey())) {
            return Integer.compare(b.getKey(), a.getKey());
        }
        return Integer.compare(b.getValue(), a.getValue());
    }
}

class Solution {
    public int maximumSum(int[] nums) {
        int n = nums.length;
        PriorityQueue<Pair<Integer, Integer>> pq = new PriorityQueue<>(new CustomComparator());
        int count = 0;
        for(int i = 0; i < n; i++){
            int temp = nums[i];
            count = 0;
            while (temp > 0){
                count += (temp % 10);
                temp /= 10;
            }
            pq.add(new Pair(count, nums[i]));
        }
        Pair<Integer, Integer> pair1 = pq.remove();
        count = -1;
        while(!pq.isEmpty()){
            Pair<Integer, Integer> pair2 = pq.remove();
            if(pair1.getKey().equals(pair2.getKey())){
                if(pair1.getValue() + pair2.getValue() > count) count = pair1.getValue() + pair2.getValue();
            }
            pair1 = pair2;
        }
        return count;
    }
}

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumSum = function(nums) {
    check = {}
    let n = nums.length
    let count = 0
    let temp = 0
    let result = -1
    for(let i = 0; i < n; i++){
        count = 0
        temp = nums[i]
        while(temp > 0){
            count += (temp % 10)
            temp = Math.floor(temp /= 10)
        }
        if (check[count] == undefined){
            check[count] = nums[i]
        }
        else {
            if (result < nums[i] + check[count]) result = nums[i] + check[count]
            if (nums[i] > check[count]) check[count] = nums[i]
        }
    }
    return result;
};